File: python/01_basic_chat.py
"""Basic chat with the official tonia client.

    export TONIA_API_KEY=tonia_sk_…
    python 01_basic_chat.py
"""

from __future__ import annotations

import json
import os

from tonia import Tonia

with Tonia(
    api_key=os.environ["TONIA_API_KEY"],
    default_headers={
        "HTTP-Referer": "https://github.com/tonia-router/sdk-examples",
        "X-Tonia-Title": "sdk-examples",
    },
) as client:
    listed = client.models.list()
    ids = [model["id"] for model in listed["data"]]
    if not ids:
        raise SystemExit("empty allowlist — do not guess a model id")
    completion = client.chat.completions.create(
        model=ids[0],
        messages=[{"role": "user", "content": "Bonjour — une phrase courte."}],
    )
    print(json.dumps(completion, indent=2, ensure_ascii=False))

File: typescript/01-basic-chat.ts
/**
 * Basic chat with the official @tonia-router/sdk client.
 *
 *   export TONIA_API_KEY=tonia_sk_…
 *   npx tsx 01-basic-chat.ts
 */
import { Tonia } from "@tonia-router/sdk";

const client = new Tonia({
  apiKey: process.env.TONIA_API_KEY,
  defaultHeaders: {
    "HTTP-Referer": "https://github.com/tonia-router/sdk-examples",
    "X-Tonia-Title": "sdk-examples",
  },
});

const { data } = await client.models.list();
if (!data[0]) {
  throw new Error("this key has no models; check the profile allowlist in the portal");
}

const completion = await client.chat.completions.create({
  model: data[0].id,
  messages: [{ role: "user", content: "Bonjour — une phrase courte." }],
});

console.log(JSON.stringify(completion, null, 2));
